extensions/amp-video-docking/0.1/controls.js

```diff
@@ -320,11 +325,19 @@ export class Controls {
 
   /** @private */
   showOnTapOrHover_() {
-    const onTapOrHover = () => this.show_();
     const {overlay} = this;
 
-    this.listenWhenEnabled_(overlay, 'mouseup', onTapOrHover);
-    this.listenWhenEnabled_(overlay, 'mouseover', onTapOrHover);
+    this.listenWhenEnabled_(overlay, 'click', () => {
+      this.show_();
+    });
+
+    this.listenWhenEnabled_(overlay, 'mouseover', () => {
+      // Show on hover only on mouse devices to prevent click/hover passthru.
+      if (!this.ampdoc_.getBody().classList.contains('amp-mode-mouse')) {
```

Comment:
the whole `amp-mode-mouse/touch` thing is faulty. Fails badly on touch enabled laptops. `mouseover` should't be triggered from any touch interaction anyway, I am not sure I understand what this check fixes.

Reply:
It seems like `mouseover` is triggered *on tap* for mobile on Chrome, which causes the controls to be shown.

[Reduced example here.](https://jsbin.com/rohesijowi/1/edit?output). If you look at it using device mode from Chrome devtools you'll see that both `mouseover` and `click` are triggered in the same event cycle. I don't know if this is a bug or some weird intended behavior :) `click` should not be triggered at all without its layer being visible first.

Reply:
Per offline discussion this can also be worked around using `requestAnimationFrame`. Also added an immediate hide trigger on mouseout, makes it a bit nicer. PTAL.